src/controllers/HistoryController.ts

```diff
@@ -44,19 +46,23 @@ export class HistoryController {
     response.send(html);
   };
 
-  private latestHistory = async (request: Request, response: Response): Promise<void> => {
-    const records = await this.fetcherHistoryRepository.latest();
+  private latestHistory = async (_request: Request, response: Response): Promise<void> => {
+    const records = await this.priceDataRepository.latest();
     response.send(records);
   };
 
   private symbolHistory = async (request: Request, response: Response): Promise<void> => {
-    const records = await this.fetcherHistoryRepository.latestSymbol(request.params.symbol);
+    const result = this.feedSymbolChecker.apply(request.params.symbol);
+    if (!result) return;
+
+    const [feedBase, feedQuote] = result;
+    const records = await this.priceDataRepository.latestPrice(feedBase, feedQuote);
 
     response.send(
       records.map((r) => {
         return {
           fetcher: r.fetcher,
-          symbol: r.symbol,
+          symbol: r.feedBase + '-' + r.feedQuote,
```

Comment:
looks like this PR will not be working if you not fix this base/quote, so let's merge the fix before we methe this one

Reply:
Well, the `PriceDataRepository` is independent of that.
In this case, the MetalPriceFetcher performs an adaptation of the symbol before passing the price to the repository.

In this test that I did I can see the right prices with the right symbols on the history api:

```
[{"_id":"667e9ef38b7c7911c45621fd","fetcher":"GoldApiPrice","value":"75.0734","valueType":"Price","timestamp":1719574259,"feedBase":"USD","feedQuote":"XAU","fetcherSource":"","__v":0},{"_id":"667e9eef8b7c7911c45621f9","fetcher":"MetalPriceApi","value":"75.06692851358483","valueType":"Price","timestamp":1719574255,"feedBase":"USD","feedQuote":"XAU","fetcherSource":"","__v":0},{"_id":"667e9eef8b7c7911c45621f6","fetcher":"MetalsDevApi","value":"75.0694","valueType":"Price","timestamp":1719574255,"feedBase":"USD","feedQuote":"gold","fetcherSource":"","__v":0},
```